scripts/pre-commit

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+set -o errexit
+
+exit_() {
+    echo ""
+    echo "$1"
+    echo ""
+    echo "This hook can be skipped if needed with 'git commit --no-verify'"
+    echo "See '.git/hooks/pre-commit', installed from 'scripts/pre-commit'"
+    exit 1
+}
+
+if [[ $(git config --get user.name | wc -w) -lt 2 ]]; then
+    # This heuristic avoids bad user names such as "root" or "Ubuntu"
+    # or a computer login name. A full name should (usually) have at
+    # least two words. We can change this if needed later.
+    exit_ "Commit failed: please fix your Git user name (see docs/Contributing.md)"
+fi
+
+if ! git diff-index --check --cached HEAD --; then
+    exit_ "Commit failed: please fix the conflict markers or whitespace errors"
+fi
+
+mapfile -t files < <(git diff --cached --name-only --diff-filter=ACMR)
```

Comment:
The macOS default bash isn't new enough to support mapfile.

One solution is to require people to put a newer bash on the path in front of the default bash.  However, I have come across instances where apps like VSCode don't pay attention to the .bashrc when doing a commit - and so this fails using a UI.

Another would be to assume that if the `mapfile` fails then we don't do the early exit, and just take the potentially lengthy path.

I'm not sure I have a good suggestion here.

Reply:
Add a baseline .editorconfig file. Was able to get everything except for the line length part of it.

I think we can create a version of the pre-commit script that doesn't depend on mapfile.

Reply:
I think I have update the scripts to work correctly with out mapfile.